osom_converter/create_volume_viewer_data.py

```diff
@@ -0,0 +1,160 @@
+from netCDF4 import Dataset
+import typer
+import numpy as np
+import os
+from datetime import datetime, timedelta
+from osom_converter import scoord_du_new2 as scoor_du
+
+app = typer.Typer()
+
+
+def fprintf(stream, format_spec, *args):
+    stream.write(format_spec % args)
+
+
+@app.command()
+def createOsomData(
+    osom_gridfile: str, osom_data_file: str, output_folder: str, data_descriptor: str
+):
+
+    """
+    Converts NETCDF files (*.nc) form the osom model to data files that can be read
+    in the volume viewer desktop app
+
+    osom_gridfile: grid file providing uv transformation coordinates ( i.e: file provided by this tool osom_grid4_mindep_smlp_mod7.nc)
+    osom_data_file: osom data file with multi-variable data
+    output_folder: location where the resulting data will be saved
+    data_descriptor: variable to extract from the osom data file (temp, salt)
+
+    """
+    # Default factors/scalers
+    verticalLevels = 15
+    downscaleFactor = 2
+
+    # Read files
+    typer.echo(" Reading " + osom_gridfile)
+    nc_grid_values = Dataset(osom_gridfile, "r")
+    typer.echo(" Reading " + osom_data_file)
+    nc_dataFile = Dataset(osom_data_file, "r")
+
+    # Assigning grid variables
+    h = nc_grid_values.variables["h"][:]
+    x_rho = nc_grid_values.variables["x_rho"][:]
+
+    # Assigning data variables
+    typer.echo(" Assigning data from data files")
+    data = nc_dataFile.variables[data_descriptor][:]
+    zeta = nc_dataFile.variables["zeta"][:]
+    vtransform = nc_dataFile.variables["Vtransform"][:]
+    vstretching = nc_dataFile.variables["Vstretching"][:]
+    theta_s = nc_dataFile.variables["theta_s"][:]
+    theta_b = nc_dataFile.variables["theta_b"][:]
+    hc = nc_dataFile.variables["hc"][:]
+    ocean_time = nc_dataFile.variables["ocean_time"][:]
+
+    # Compute and plot ROMS vertical stretched coordinates
+    typer.echo(" Computing ROMS vertical stretched coordinates")
+    z, s, C = scoor_du.scoord_du_new2(
+        h,
+        zeta[0, :, :],
+        vtransform,
+        vstretching,
+        theta_s,
+        theta_b,
+        hc,
+        verticalLevels,
+        0,
+    )
+
+    min_z = np.floor(z.min())
+    max_z = np.ceil(z.max())
+
+    # set nbSlices (per meter)
+    query_depths = np.arange(min_z, max_z + 1)
+    query_depths_shape = np.shape(query_depths)
+    slices = query_depths_shape[0]
+
+    # getting minimum and max data
+    min_data = np.floor(data.min())
+    max_data = np.ceil(data.max())
+
+    # scale data to the range of 0 and 1
+    data = (data - min_data) / (max_data - min_data)
```

Comment:
The logic works me.  `data` is between [0,1]. Ceiling and floor operations are to get a close interval without high number decimals and/or values over 1. Let me know if that answers the question.

Reply:
Just to make sure I understand correctly... The data before before it's overwritten is floating point array. Say that max of the data is 9.2. Ceil will make `max_data` = 10 and that value is used in the scaling. Therefore the upper endpoint of the scaled data is scaled to the value 10 and not the true max of the data (9.2). This would also happen for the minimum end of the range. Just wanted to make sure that difference was trivial. Does that make sense?

Reply:
I see your point, and until now I am aware of this detail. Graphically, it would mean that  no pixel will have the maximum color intensity ( i.e color space 8 bit = 255 ). Taking the floor of the maximum would mean that the most intense pixel is not the real maximum of the dataset.  
Let's say the maximum value is 9.238456 and the prior number close to the maximum is 9.23333. Taking the floor of the maximum would make us lose those two values and all of the other numbers between the range [ 9 ,  9.238456 ]. Taking the maximum's ceiling includes the values in that range, but again the maximum pixel intensity in the frame would be 248 (in the range 0-255).
TLDR:  Taking the maximum's ceiling can give us small graphic artifacts, but at least we wont be losing data.

Reply:
Great! Yeah that makes sense.